### ISSUE TITLE
Click on the debug option closes the menu

### ISSUE THREAD

**Issue by igorskuhar** [godotengine/godot#7262]
**Operating system or device - Godot version:**
Mac OS X 10.11.4 (MBP 2015 Retina) - Godot v2.1.1.stable

**Issue description** (what happened, and what was expected):
Each click on the debug option to select/deselect it, closes the menu.
So, in order to select four debug options I have to reopen the "Debug options" menu four times.

<img width="245" alt="debug_option_click_closes_menu" src="https://cloud.githubusercontent.com/assets/5592468/20950450/150e0e50-bc20-11e6-80ee-046652e5b91a.png">

**Steps to reproduce:**
1. Open the "Debug options" menu.
2. Click on any of the options to select/deselect it.
3. Menu will close.

**Comment by Hinsbart** [comment]
But isn't this a kinda standard behaviour? I think that these types of toolbar menus close on selection in most of the applications I use.
So if we change it, someone else will come and complain that he/she has to click somewhere else on the screen to actually close the menu...
Then how about making you have to use this menu less often instead?
By having the editor remember debug settings on a per-project basis so you don't need to set them each time you open the project.

**Comment by igorskuhar** [comment]
@Hinsbart 
Can you give an example of other application behaving in the same manner?
I think that if it was an ordinary context menu with a list of different actions AND checkboxes, then close on click would be expected.
But in this case, Debug options menu should act more like multiselect dropdown, so you can select all the options you need before you decide to close it by clicking an icon again or anywhere outside of the menu.

I have created another issue before about remembering the selection, please take a look at #7261.

**Comment by Zylann** [comment]
So if you want that menu to keep open, then it should not be a menu, but a window with options in it, otherwise it will be inconsistent with standard behaviour.
I don't know any other application having a drop-down menu that stays open after clicking on a checkable item (at least on Windows and Linux).

**Comment by igorskuhar** [comment]
@Zylann 
> I don't know any other application having a drop-down menu that stays open after clicking on a checkable item (at least on Windows and Linux).

You are probably talking about **single select** dropdowns, and what we have here is **multi select**.
Think about it. Let's say it has 10 options instead of 6, and you want to select 9 of them at once.
Currently, you would have to open the menu 6 times and select item 6 times, that's 12 clicks.
If menu stayed there until you decide to click outside or on the icon, that would be 7 clicks to select all the options.
Basically the more options the menu has, the more cumbersome the process becomes.

**Comment by Zylann** [comment]
I agree with the amount of items, and the fact a menu could not close if it has checkable items.
My point is, it should be clear for the user that one kind of menu closes and another doesn't.

Hmm I'll amend what I said about "no other applications do that", because I just found another menu in Visual Studio that stays open when you check an item:

![vs_add_remove_button](https://cloud.githubusercontent.com/assets/1311555/20968854/49a6cc0c-bc7e-11e6-8ff7-77783ec21a00.png)

But again, this menu is specific and contains only checkable items. I don't think you'll find that in mixed menus.

**Comment by igorskuhar** [comment]
> But again, this menu is specific and contains only checkable items. I don't think you'll find that in mixed menus.

Agreed. 👍 
Our "Debug options" menu also has only checkable items.

**Comment by Jylhis** [comment]
Tested on Windows 10, godot v2.1.2.stable and this seems to be fixed. Probably in this #7306

**Comment by bojidar-bg** [comment]
>  If we do that, I would also ask that we do this for the snap settings of the TextureRegion editor, and also Edit menu of the scene (though there are mixed types of items in this one :s)

Not completely fixed yet, I guess (though it should be junior-job from now on).
